Bourreau/app/models/bourreau_worker.rb

```diff
@@ -536,4 +546,120 @@ def check_for_tasks_stuck_in_ruby
     end
   end
 
+  # Handles new tasks submitted by "task". To submit a new task, a
+  # task must create a new-task-*.json JSON file at the root of its
+  # work directory.  Once a JSON file has been handled, it is deleted.
+  #
+  # Method parameter:
+  # * "task" is the submitter task. JSON files will be searched in the
+  # work dir of "task".  New tasks are submitted on the same Bourreau
+  # than "task".  New tasks belong to the same user than "task".
+  # 
+  def handle_tasks_submitted_by task
+    workdir = task.full_cluster_workdir
+    return if workdir.blank? # in case workdir doesn't exist yet
+    Dir.glob(File.join(workdir,"new-task-*.json")).each do |filename|
+      worker_log.info("Found new task file: #{filename}.")
+      begin
+        file = File.open(filename, "r")
+        if(file.flock(File::LOCK_NB|File::LOCK_EX)) # Here we ensure that only 1 worker can access the file at
+                                                    # the same time. Otherwise, the new task may be submitted
+                                                    # by several workers. See nice examples at:
+                                                    # http://www.codegnome.com/blog/2013/05/26/locking-files-with-ruby
+          file_content = file.read
+          submit_task_from_string(file_content,task)
+          file.close # This also releases the lock. 
+          File.delete(filename)
+        else
+          worker_log.info("Another worker is working on this file: ignoring it.")
+          file.close # In case the lock was not obtained, still close the file. 
+        end
+      rescue => ex
+        worker_log.info("Error while submitting new task: #{ex.message}.")
+        File.delete(filename) if File.exists?(filename)
+      end
+    end
+  end
+
+  # Creates and submits a task defined by a JSON object.
+  # Parameters:
+  # * json_string: A string containing a JSON object defining the task.
+  #                Example:
+  #                  {
+  #                    "tool-class": "CbrainTask::TestTool",
+  #                    "description": "A task running TestTool",
+  #                    "parameters": {
+  #                       "important_number": "123",
+  #                       "dummy_paramet4er": "432"
+  #                    }
+  #                  }
+  # * current_task: The current task from which the new task is submitted.
+  #                 Used to set the user, bourreau id and results data provider id
+  #                 for the new task.
+  # 
+  def submit_task_from_string json_string,current_task
+
+    # Parses JSON string and checks format
+    validate_json_string(json_string) # Raises an exception if string is not valid
+    new_task_hash = JSON.parse(json_string)
+    worker_log.info("Submitting task #{new_task_hash}")
+    
+    # Creates task
+    task_class_name = new_task_hash["tool-class"]
+    new_task        = CbrainTask.const_get(task_class_name).new # Raises an exception if tool class is not found
```

Comment:
Not strict enough, you can do

```ruby
CbrainTask.const_get('Kernel')
```
and get any class completely outside the CbrainTask:: namespace ...